src/aws-cpp-sdk-core/source/utils/stream/ResponseStream.cpp

```diff
@@ -43,12 +50,26 @@ ResponseStream& ResponseStream::operator=(ResponseStream&& toMove)
     }
 
     ReleaseStream();
+    toMove.DeregisterStream();
     m_underlyingStream = toMove.m_underlyingStream;
     toMove.m_underlyingStream = nullptr;
+    RegisterStream();
 
     return *this;
 }
 
+Aws::IOStream& ResponseStream::GetUnderlyingStream() const
+{
+    if (!m_underlyingStream)
+    {
+        assert(m_underlyingStream);
```

Comment:
Should we log first and fail assert after during debug?

Reply:
They are equal in messaging to the developer.
Except FATAL tracing is just a last resort if library is built in Release mode.
It is a really just "not crash", the next line will highly likely produce a memory leak reported by custom allocator set in the API Handle.